### PR TITLE
chore: Add a company name field to the issue templates

### PR DESCRIPTION
**Pull request by sfc-gh-jmichalak**
As discussed internally, this change adds a new `Company Name `field to the issue templates.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/01-bug.yml [hunk 1]
```diff
@@ -249,6 +249,12 @@ body:
         - *SF_TF_GOSNOWFLAKE_LOG_LEVEL=debug* - if you feel the problem may be connected with the Snowflake driver, please set this one for additional driver logs.
       placeholder: https://gist.github.com/example/12345678
 
+  - id: company-name
```
Comment (sfc-gh-asawicki) [inline .github/ISSUE_TEMPLATE/01-bug.yml:252]: Two remarks:
1. let's put it much closer to the top (maybe even in the `Terraform CLI and Snowflake Provider Versions` section)
2. Let's rephrase to: `Optional field. Providing this information allows us to communicate faster which may lead to faster issue resolution.`

**Review by sfc-gh-asawicki — CHANGES_REQUESTED** [review]

File: .github/ISSUE_TEMPLATE/01-bug.yml (hunk 1 above)
Comment (sfc-gh-jmichalak) [reply .github/ISSUE_TEMPLATE/01-bug.yml]: Done both.

**Review by sfc-gh-asawicki — APPROVED** [review]

File: .github/ISSUE_TEMPLATE/04-feature-request.yml [hunk 2]
```diff
@@ -19,6 +19,12 @@ body:
         - [Guide on creating issues + FAQ + Commonly Known Issues](https://github.com/Snowflake-Labs/terraform-provider-snowflake/blob/main/CREATING_ISSUES.md) - If you would like to create a GitHub issue, please read our guide first. It contains useful links, FAQ, and commonly known issues with solutions that may already solve your case.
         - [Provider Roadmap](https://github.com/Snowflake-Labs/terraform-provider-snowflake/blob/main/ROADMAP.md)
 
+  - id: company-name
+    type: input
+    attributes:
+      label: Company Name
+      description: Optional field. Providing this information allows us to communicate faster which may lead to faster issue resolution.
```
Comment (sfc-gh-jcieslak) [inline .github/ISSUE_TEMPLATE/04-feature-request.yml:26]: Don't know if in all templates we should refer to faster resolution of "issue" in e.g. feature request. But, other than that 👍.

Comment (sfc-gh-asawicki) [reply .github/ISSUE_TEMPLATE/04-feature-request.yml:26]: feature request is also a GH issue, I think it's fine

Comment (sfc-gh-jmichalak) [reply .github/ISSUE_TEMPLATE/04-feature-request.yml:26]: Why not? In this case, resolved issue == implemented feature.

**Pull request merged**